Mastermind/Domain/BusinessRules/GamePlay.cs

```diff
@@ -0,0 +1,72 @@
+using Mastermind.Domain.Models;
+using Mastermind.Randomizer;
+
+namespace Mastermind.Domain.BusinessRules
+{
+    public class GamePlay
+    {
+        private readonly IRandomizer _randomizer;
+        
+        public GamePlay(IRandomizer randomizer)
+        {
+            _randomizer = randomizer;
+        }
+
+        public Game SetupGame()
+        {
+            return new Game
+            {
+                SelectedColours =_randomizer.GetRandomColours(GameConstants.SelectedNumberOfColours),
+                GuessingCount = 0,
+                GameState = GameStatus.Playing
+            };
+        }
+        
+        public void EvaluatePredictedAnswer(Game game)
+        {
+            UpdateGameLostStatus(game);
+            
+            game.GuessingCount++;
+            
+            UpdateCluesAccordingThePrediction(game);
+            game.Clues = GetShuffledClues(game.Clues);
+
+            UpdateGameWonStatus(game);
+        }
+
+        private List<Clue> GetShuffledClues(List<Clue> clues)
+        {
+            return _randomizer.GetShuffledArray(clues);
+        }
+
+        private void UpdateCluesAccordingThePrediction(Game game)
+        {
+            game.Clues?.Clear();
+            var predictedAnswer = game.LatestPlayerGuess;
+            
+            for (var index = 0; index < game.SelectedColours.Length; index++)
+            {
+                var selectedColour = game.SelectedColours[index];
+                
+                if (!predictedAnswer.Contains(selectedColour)) continue;
+
+                game.Clues?.Add(predictedAnswer[index] == selectedColour ? Clue.Black : Clue.White);
+            }
+        }
+        
+        private void UpdateGameWonStatus(Game game)
+        {
+            if (game.Clues?.Count(c => c == Clue.Black) == 4)
```

Comment:
`4` here is a magic number, since it isn't entirely clear what it represents. Using a constant here could give it more meaning.